### PR TITLE
edge 1018 - adding a webhooks page to webrtc

### PR DESCRIPTION
**Pull request by smckinnon-bw**
addition of a page in the new doc-site to cover the callbacks for POST .../participants, to ensure that the events were articulated, and that the beta status was documented.

### CODE REVIEW

File: site/docs/webrtc/webhooks.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  - webhook
+image: ../../static/img/bandwidth-logo.png
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+## Subscribing to Webhooks
+
+The only API call that allows the registration of a callback / webhook is the POST to the participants resource that is used to create a new participant. The callback/webhook URL is declared in the `POST` to the `https://api.webrtc.bandwidth.com/v1/accounts/{accountId}/participants` endpoint. If the `callbackUrl` property is included in the payload, the value associated with that property is assumed to be a URL. That URL (if found to be valid) will be invoked on events that occur as part of the Participant life cycle.
```
Comment (carrington-bw) [inline site/docs/webrtc/webhooks.md:19]: I would just refer to it as createParticipant endpoint, as it is labeled in the API Docs

File: site/docs/webrtc/webhooks.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+:::note
+Participant event webhooks are currently in Beta, and must be activated by submitting a support ticket to Bandwidth.
```
Comment (carrington-bw) [inline site/docs/webrtc/webhooks.md:22]: Perhaps include "(Beta)" in the title of the page and in the nav tree

**Review by psherk-bw — APPROVED** [review]

File: site/docs/webrtc/webhooks.md (hunk 1 above)
Comment (smckinnon-bw) [reply site/docs/webrtc/webhooks.md]: done

File: site/docs/webrtc/webhooks.md (hunk 2 above)
Comment (smckinnon-bw) [reply site/docs/webrtc/webhooks.md:24]: I added Beta to the internal title.  I was not a fan of exposing it in the sidebar.

File: site/docs/webrtc/webhooks.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+- 'onLeave' - loss of the websocket connection, possibly due to loss of IP connectivity.
+
+## Request Payload
+
+The callback URL will be invoked by Bandwidth when the above events are detected. It will result in a `POST` to the URL specified as the value associated with the `callbackUrl` parameter. The Request Payload parameters that will be included in the `POST` body are...
+
+| Property      | Description                                                                                                       |
+| :------------ | :---------------------------------------------------------------------------------------------------------------- |
+| event         | one of `onConnect`, `onDisconnect` or `onLeave`                                                                   |
+| timestamp     | a Timestanp of the event, in milliseconds                                                                         |
```
Comment (hamermike) [inline site/docs/webrtc/webhooks.md:46]:
```suggestion
| timestamp     | a Timestamp of the event, in milliseconds                                                                         |
```

**Review by hamermike — CHANGES_REQUESTED** [review]

**Review by hamermike — APPROVED** [review]

File: site/docs/webrtc/webhooks.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+:::note
+Participant event webhooks may change in the future. All attempts will be made to retain the syntax and semantics of the current implementation, but until Bandwidth completes the process of learning with our customers that the implementation is optimal, we cannot assure it. Bandwidth welcomes any and all comments on the implementation while in the Beta phase.
+:::
+
+## Life Cycle Events
+
+The events that will trigger invocation of the specified URL are:
+
+- 'onConnect' - establishment of the websocket connection that will allow the flow of WebRTC Media, an explicit action on the part of the websocket client.
```
Comment (carrington-bw) [inline site/docs/webrtc/webhooks.md:35]: I think we should be somewhat vague in these descriptions. For example refer to the connection as a signalling connection, not a web socket connection. 

These callbacks represent individual device events (onConnect is a single device establishing an authenticated signaling connection, not the flow of media)

**Pull request merged**